File: Eugenics/Eugenics/Models/Class.cs
using System.Collections.Generic;

namespace Eugenics.Models
{
    public class Class
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int MaxHP { get; set; }
        public int MaxStr { get; set; }
        public int MaxMag { get; set; }
        public int MaxSkl { get; set; }
        public int MaxSpd { get; set; }
        public int MaxLck { get; set; }
        public int MaxDef { get; set; }
        public int MaxRes { get; set; }
        public int PairStr { get; set; }
        public int PairMag { get; set; }
        public int PairSkl { get; set; }
        public int PairSpd { get; set; }
        public int PairLck { get; set; }
        public int PairDef { get; set; }
        public int PairRes { get; set; }
        public int PairMov { get; set; }
        public bool IsBaseClass { get; set; }
    }
}

File: Eugenics/Eugenics/Scripts/models/class.js
(function () {
    "use strict";

    define([
        "knockout"
    ], function (ko) {
        var ClassViewModel = function (classObj) {
            this.id = ko.observable(classObj.id);
            this.name = ko.observable(classObj.name);
            this.maxHP = ko.observable(classObj.maxHP);
            this.maxStr = ko.observable(classObj.maxStr);
            this.maxMag = ko.observable(classObj.maxMag);
            this.maxSkl = ko.observable(classObj.maxSkl);
            this.maxSpd = ko.observable(classObj.maxSpd);
            this.maxLck = ko.observable(classObj.maxLck);
            this.maxDef = ko.observable(classObj.maxDef);
            this.maxRes = ko.observable(classObj.maxRes);
            this.pairStr = ko.observable(classObj.pairStr);
            this.pairMag = ko.observable(classObj.pairMag);
            this.pairSkl = ko.observable(classObj.pairSkl);
            this.pairSpd = ko.observable(classObj.pairSpd);
            this.pairLck = ko.observable(classObj.pairLck);
            this.pairDef = ko.observable(classObj.pairDef);
            this.pairRes = ko.observable(classObj.pairRes);
            this.pairMov = ko.observable(classObj.pairMov);
            this.isBaseClass = ko.observable(classObj.isBaseClass);
        };

        return ClassViewModel;
    });
})();
